### PR TITLE
update report.n_draws when slicing

### PR DESCRIPTION
**Pull request by aloctavodia**
When slicing a trace the report property is lost, this add it back. With this fix all the properties will set to 
`None`. The `n_draws` values will be updated to the length of the slice. We could update the other values too, but most of them, if not all of them, are related to the entire trace and not the slice so probably it is better to leave them as `None`.

### CODE REVIEW

File: pymc3/backends/ndarray.py [hunk 1]
```diff
@@ -54,13 +55,12 @@ def save_trace(trace: MultiTrace, directory: Optional[str]=None, overwrite=False
     str, path to the directory where the trace was saved
     """
     warnings.warn(
-        'The `save_trace` function will soon be removed.'
-        'Instead, use ArviZ to save/load traces.',
+        "The `save_trace` function will soon be removed." "Instead, use ArviZ to save/load traces.",
```
Comment (michaelosthege) [inline pymc3/backends/ndarray.py]: black messed up here: `removed. " "Instead`

File: pymc3/backends/ndarray.py [hunk 2]
```diff
@@ -97,12 +99,11 @@ def load_trace(directory: str, model=None) -> MultiTrace:
     pm.Multitrace that was saved in the directory
     """
     warnings.warn(
-        'The `load_trace` function will soon be removed.'
-        'Instead, use ArviZ to save/load traces.',
+        "The `load_trace` function will soon be removed." "Instead, use ArviZ to save/load traces.",
```
Comment (michaelosthege) [inline pymc3/backends/ndarray.py]: same as above

**Review by michaelosthege — CHANGES_REQUESTED** [review]

**Review by AlexAndorra — APPROVED** [review]
Good catch from Michael, but apart from that everything looks good now, thanks a lot @aloctavodia !
Will merge once you push the last modif and tests pass again 😊

File: pymc3/backends/base.py [hunk 3]
```diff
@@ -519,16 +518,18 @@ def get_sampler_stats(self, stat_name, burn=0, thin=1, combine=True,
...
         return _squeeze_cat(results, combine, squeeze)
 
     def _slice(self, slice):
         """Return a new MultiTrace object sliced according to `slice`."""
         new_traces = [trace._slice(slice) for trace in self._straces.values()]
         trace = MultiTrace(new_traces)
         idxs = slice.indices(len(self))
-        trace._report = self._report._slice(*idxs)
+        trace.report._chain_warnings = self.report._chain_warnings
+        trace.report._n_draws = len(trace)
```
Comment (OriolAbril) [inline pymc3/backends/base.py:532]: does this work if `save_warmup` is True?

Comment (AlexAndorra) [reply pymc3/backends/base.py:532]: I think slicing is done only when `discard_tune_sample=False`, so I'd say yes 🤷‍♂️

**Pull request closed without merging**